src/java.base/share/classes/java/util/concurrent/locks/AbstractQueuedLongSynchronizer.java

```diff
@@ -369,14 +369,19 @@ final int acquire(Node node, long arg, boolean shared,
             } else if (node.status == 0) {
                 node.status = WAITING;          // enable signal and recheck
             } else {
-                long nanos;
                 spins = postSpins = (byte)((postSpins << 1) | 1);
-                if (!timed)
-                    LockSupport.park(this);
-                else if ((nanos = time - System.nanoTime()) > 0L)
-                    LockSupport.parkNanos(this, nanos);
-                else
-                    break;
+                try {
+                    long nanos;
+                    if (!timed)
+                        LockSupport.park(this);
+                    else if ((nanos = time - System.nanoTime()) > 0L)
+                        LockSupport.parkNanos(this, nanos);
+                    else
+                        break;
+                } catch (Error ex) {            // rethrow VM errors
```

Comment:
Part of me things this should be Throwable to allow for possible runtime exceptions when timed-parking virtual threads. There is a separate work needed to ensure a runtime exception is never throw but it would at least cancel here.

Reply:
That would be okay too as there are no checked exceptions here.

Reply:
@AlanBateman If we really want it airtight I think we need to catch Throwable and sneaky-rethrow it? 🤔